File: common-account/src/main/java/tech/corefinance/account/common/model/AccountFee.java
package tech.corefinance.account.common.model;

import lombok.Data;
import tech.corefinance.product.common.enums.ProductFeeType;

@Data
public class AccountFee {
    private boolean activated;
    private String id;
    private String name;
    private ProductFeeType type;
    private Double amount;
    private String currencyId;
    private Double disbursedPercent;

    private MonthlyPayOption monthlyPayOption;
    private Boolean requiredFeeApplication;
}

File: adminweb/src/app/classes/accounts/AccountFee.ts
import { MonthlyPayOption } from "../products/Product";
import { ProductFeeType } from "../products/ProductFeeType";

export class AccountFee {
    activated = true;
    id = "";
    name = "";
    type: ProductFeeType = ProductFeeType.MONTHLY_FEE;
    amount = "";
    currencyId = "";
    disbursedPercent = 0.0;

    monthlyPayOption: MonthlyPayOption = MonthlyPayOption.MONTHLY_LAST_DAY;
    requiredFeeApplication = false;
}